src/pages/SignUp/SignUp.jsx

```diff
@@ -3,7 +3,92 @@ import Button from "../../components/Button/Button";
 import logo from "../../assets/images/logo.png";
 import InputBox from "../../components/InputBox/InputBox";
 
+import {
+  getAuth,
+  createUserWithEmailAndPassword
+} from "firebase/auth";
+import app from "../../firebase";
+import {
+  useState,
+  useEffect
+} from "react";
+
 const SignUp = ({ login }) => {
+  const [registerEmail, setRegisterEmail] = useState("");
+  const [registerPassword, setRegisterPassword] = useState("");
+  const [confirmPassword, setConfirmPassword] = useState("");
+  const [emailError, setEmailError] = useState("");
+  const [emailSuccess, setEmailSuccess] = useState("");
+  const [password1Error, setPassword1Error] = useState("");
+  const [password1Success, setPassword1Success] = useState("");
+  const [password2Error, setPassword2Error] = useState("");
+  const [password2Success, setPassword2Success] = useState("");
+
+  
+  const passRegex = /^(?=.*[0-9])(?=.*[!@#$%^&*])[a-zA-Z0-9!@#$%^&*]{6,16}$/
+  const emailRegex = /^\w+@[a-zA-Z_]+?\.[a-zA-Z]{2,3}$/
+
+  const auth = getAuth(app);
+
+const validateInputs = () => {
+  let result = true;
+  setEmailError("");
+  setEmailSuccess("");
+  setPassword1Error("");
+  setPassword1Success("");
+  setPassword2Error("");
+  setPassword2Success("");
+  if (!emailRegex.test(registerEmail)) {
+    setEmailError("Error - Your email must contain an @ symbol");
+    result = false;
+  } else {
+    setEmailSuccess("Successful email entered");
+  }
+  if (confirmPassword != registerPassword) {
+    setPassword2Error("Error - passwords do not match!");
+    result = false;
+  } else {
+    setPassword2Success("Passwords Match");
+  }
+  if (!passRegex.test(registerPassword)) {
+    setPassword1Error("Error - password must contain at least 8 characters, 1 uppercase letter, and 1 symbol");
+    result = false;
+  }
+  if (confirmPassword === registerPassword && passRegex.test(registerPassword)) {
+    setPassword1Success("Password meets requirements.")
+    result = false;
```

Comment:
Remove 'result = false' on line 59 what causes function to return false even if the inputs are correct